### PR TITLE
feat: disable reactions for ephemeral message WPB-629

### PR DESCRIPTION
**Pull request by MarcinR**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-629" title="WPB-629" target="_blank">TaskWPB-629</a>  Create context menu with reactions
  </td></table>
  <br />
 

#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/MessageActionsViewController.swift [hunk 1]
```diff
@@ -53,13 +54,13 @@ class MessageActionsViewController: UIAlertController {
     }
 
     private func addReactionsView(withDelegate delegate: ReactionPickerDelegate) {
+        guard let customContentPlaceholder = self.view.findLabel(withText: MessageActionsViewController.MessageLabelMarker),
+              let customContainer =  customContentPlaceholder.superview else { return }
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/MessageActionsViewController.swift]:
```suggestion
              let customContainer = customContentPlaceholder.superview else { return }
```

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/MessageActionsViewController.swift [hunk 2]
```diff
@@ -26,7 +26,8 @@ class MessageActionsViewController: UIAlertController {
 
     static func controller(withActions actions: [MessageAction],
                            actionController: ConversationMessageActionController) -> MessageActionsViewController {
-        let controller = MessageActionsViewController(title: MessageLabelMarker,
+        let title = actionController.canPerformAction(action: .react(.like)) ? MessageLabelMarker : nil
+        let controller = MessageActionsViewController(title: title,
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/MessageActionsViewController.swift:30]: Why do you want to create a `MessageActionsViewController` with `nil` title and message if it's `.react(.like)`? Why not make this function return an optional result?

File: wire-ios/Wire-iOS Tests/ConversationMessageCell/MessageActionsViewControllerTests.swift [hunk 3]
```diff
@@ -28,6 +28,41 @@ final class MessageActionsViewControllerTests: XCTestCase {
...
+        message.isEphemeral = true
+        let actionController = ConversationMessageActionController(responder: nil, message: message, context: .content, view: UIView())
+        // WHEN
+        let messageActionsViewController = MessageActionsViewController.controller(withActions: MessageAction.allCases,
+                                                                            actionController: actionController)
+        // THEN
+        XCTAssertFalse(messageActionsViewController.view.containsBasicReactionPicker())
+    }
+
+    func testReactionPicker_DoesnExistForFailedMessage() {
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS Tests/ConversationMessageCell/MessageActionsViewControllerTests.swift]: *DoesNot?

File: wire-ios/Wire-iOS Tests/ConversationMessageCell/MessageActionsViewControllerTests.swift [hunk 4]
```diff
@@ -28,6 +28,41 @@ final class MessageActionsViewControllerTests: XCTestCase {
...
+        let message = MockMessageFactory.textMessage(withText: "Test tests")
+        let actionController = ConversationMessageActionController(responder: nil, message: message, context: .content, view: UIView())
+        // WHEN
+        let messageActionsViewController = MessageActionsViewController.controller(withActions: MessageAction.allCases,
+                                                                            actionController: actionController)
+        // THEN
+        XCTAssertTrue(messageActionsViewController.view.containsBasicReactionPicker())
+    }
+
+    func testReactionPicker_DoesnExistForEphemeralMessage() {
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS Tests/ConversationMessageCell/MessageActionsViewControllerTests.swift]: *DoesNot?

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/MessageActionsViewController.swift (hunk 2 above)
Comment (MarcinR) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/MessageActionsViewController.swift:30]: Title label is a placeholder for Reaction Picker.  If we can't add reaction then we don't need this title label. So I set nil for a title.

Even if we don't show reaction picker we're adding UIAlertActions to that alert controller. That's why this function doesn't returning nil

**Review by KaterinaWire — APPROVED** [review]

**Pull request merged**